tests/Moq.Analyzers.Test/DisabledAnalyzerTests.cs

```diff
@@ -0,0 +1,199 @@
+using Microsoft.CodeAnalysis.Testing;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+
+using SetupVerifier = Moq.Analyzers.Test.Helpers.AnalyzerVerifier<Moq.Analyzers.SetupShouldBeUsedOnlyForOverridableMembersAnalyzer>;
+using SealedVerifier = Moq.Analyzers.Test.Helpers.AnalyzerVerifier<Moq.Analyzers.NoSealedClassMocksAnalyzer>;
+using SetupConfigVerifier = Moq.Analyzers.Test.Helpers.ConfigAnalyzerVerifier<Moq.Analyzers.SetupShouldBeUsedOnlyForOverridableMembersAnalyzer>;
+using SealedConfigVerifier = Moq.Analyzers.Test.Helpers.ConfigAnalyzerVerifier<Moq.Analyzers.NoSealedClassMocksAnalyzer>;
+
+namespace Moq.Analyzers.Test;
+
+/// <summary>
+/// Tests to verify that analyzers are properly disabled when configured to be disabled.
+/// This ensures that no false warnings are generated when users explicitly disable analyzers.
+/// 
+/// Performance note: When analyzers are disabled via configuration (severity = none), 
+/// the Roslyn framework automatically avoids calling the analyzer's analysis methods entirely,
+/// which provides optimal performance by avoiding unnecessary analysis work.
+/// 
+/// Test Coverage:
+/// - Pragma warning directives (#pragma warning disable/restore)
+/// - SuppressMessage attributes at method and assembly level
+/// - Configuration-based disabling via .editorconfig files
+/// - Control tests to ensure analyzers work when not disabled
+/// </summary>
+public class DisabledAnalyzerTests
+{
+    /// <summary>
+    /// Test data that provides both old and new Moq reference assembly groups
+    /// to ensure disabled analyzer behavior works across different Moq versions.
+    /// </summary>
+    public static IEnumerable<object[]> TestData()
+    {
+        return new object[][]
+        {
+            ["Test with pragma warning disable"],
+        }.WithMoqReferenceAssemblyGroups();
+    }
+
+    [Theory]
+    [MemberData(nameof(TestData))]
+    public async Task ShouldNotReportSetupDiagnosticsWhenDisabledWithPragmaWarning(string referenceAssemblyGroup, string testName)
+    {
+        string source = """
+                        public class SampleClass
+                        {
+                            public int Property { get; set; }
+                        }
+
+                        internal class UnitTest
+                        {
+                            private void Test()
+                            {
+#pragma warning disable Moq1200
```

Comment:
Failing build:

Error: /home/runner/work/moq.analyzers/moq.analyzers/tests/Moq.Analyzers.Test/DisabledAnalyzerTests.cs(53,1): error CS8999: Line does not start with the same whitespace as the closing line of the raw string literal. [/home/runner/work/moq.analyzers/moq.analyzers/tests/Moq.Analyzers.Test/Moq.Analyzers.Test.csproj]
Error: /home/runner/work/moq.analyzers/moq.analyzers/tests/Moq.Analyzers.Test/DisabledAnalyzerTests.cs(98,1): error CS8999: Line does not start with the same whitespace as the closing line of the raw string literal. [/home/runner/work/moq.analyzers/moq.analyzers/tests/Moq.Analyzers.Test/Moq.Analyzers.Test.csproj]